components/common/src/dummy_connector.rs

```diff
@@ -24,8 +24,10 @@ impl<A, O> DummyConnector<A, O> {
     }
 }
 
-// TODO: Why didn't the automatic #[derive(Clone)] works for DummyListener?
-// Seemed like it had a problem with having config_receiver inside ListenRequest.
+// #[derive(Clone)] does not work for DummyListener when compiling index_client
+// Seems like it has a problem with having config_receiver inside ListenRequest.
+// O is Option<(Sender<IndexClientToServer>, Receiver<IndexServerToClient>)>
```

Comment:
But DummyConnector contains only one field, which is a `Sender<_>`, so it must be clone.
I still wonder why can't the compiler recognize this.

Reply:
From Rust docs:

> For a generic struct, `#[derive]` implements `Clone` conditionally by adding bound `Clone` on generic parameters.

Reply:
I see! Maybe in the future it will be possible for the compiler team to make this restriction more flexible.